### PR TITLE
Fix sub path support

### LINKED ISSUE #36: PAPERLESS_FORCE_SCRIPT_NAME does not work.
This is required to run Paperless on a sub path, i.e. localhost:8000/paperless/.

This is generally required if you wish to use a proxy server with many different services.

Right now, this does not work.

As for the front end, the src/documents/static/index.html contains a `<base href="/">`, which has to be adapted to href="/paperless" or similar.

https://github.com/jonaswinkler/paperless-ng/blob/59bc467c508f2e9576d39219dc3226f095b5a054/src/documents/templates/index.html#L10

This is easy since the index page is served as a django view and template context is available. See 

https://github.com/jonaswinkler/paperless-ng/blob/59bc467c508f2e9576d39219dc3226f095b5a054/src/documents/views.py#L67

The front end further has some static assets for logos, which are always pointing towards `<host>/assets/image.png`. This is unsolved right now.

Furthermore, login/logout paths have to respect this path, as well as the admin/ link on the front end.

I don't need this, but if anyone cares about this, feel free to look into it.

### PR DESCRIPTION
**Pull request by puuu**
This commit aims to fix #36.

After setting e.g.: `PAPERLESS_FORCE_SCRIPT_NAME="/paperless"` everything should be accessible via http://localhost:8000/paperless/ . This allows paperless to run behind a proxy under a sub path.

IMHO, the last commit on file `src/paperless/urls.py` looks a bit hacky, I am open for suggestions.

### CODE REVIEW

**Comment by jonaswinkler** [comment]
First of all, thank you.

I've tried `PAPERLESS_FORCE_SCRIPT_NAME="/paperless"`, however all of the URLs now return 404 errors with the django development server.

**Comment by jonaswinkler** [comment]
With `PAPERLESS_DEBUG=True` and `PAPERLESS_FORCE_SCRIPT_NAME` set, none of the urls work. Not a dealbreaker, but do you have any idea why that happens?

File: src/paperless/urls.py [hunk 1]
```diff
@@ -35,6 +37,9 @@
 api_router.register(r"tags", TagViewSet)
 api_router.register(r"saved_views", SavedViewViewSet)
 
+base_path = ""
+if settings.BASE_URL:
+    base_path = settings.BASE_URL[1:] + "/"
```
Comment (jonaswinkler) [inline src/paperless/urls.py:42]: Would you mind adding a system check that verifies that FORCE_SCRIPT_NAME starts with a "/" and does not end with a "/"?

**Comment by puuu** [comment]
I only tested within the docker environment. At least for the `docker-compose.env` it must be `PAPERLESS_FORCE_SCRIPT_NAME=/paperless` (without quotes). Sorry I wrote it wrong in the first comment.

I have no experience with Django and somehow I think the solution in `src/paperless/urls.py` seems to be wrong with the additional `path()` function, but in the docker environment `FORCE_SCRIPT_NAME` seems to have no effect at all.

After digging a little bit around, I think it might be because of using uvicorn. At least [here](https://github.com/Midnighter/fastapi-mount/tree/root-path) is described that we need to set `root_path`.

May be, the Django development server respect `FORCE_SCRIPT_NAME` and that is why you get the duplicates.

If you do not mind, I will try to get it working with uvicorn and `root_path`and may rewrite the pull request.

File: src/paperless/urls.py (hunk 1 above)
Comment (puuu) [reply src/paperless/urls.py:42]: What do you mean by "system check"? Should it print a warning or raise an exception? In `src/paperless/urls.py` or `src/paperless/settings.py` ?

**Comment by puuu** [comment]
As I mentioned before, uvicorn is not aware of `FORCE_SCRIPT_NAME`. For uvicorn, the `root_path` must be set. In order to do this from gunicorn, we need to [make a subclass of UvicornWorker](https://www.uvicorn.org/deployment/#gunicorn).

I did this in `src/paperless/workers.py`.  After this I was able to make a cleaner pull request. Probably, this should now also work with the django development server (untested).

**Comment by jonaswinkler** [comment]
Everything seems to work. Thanks!

**Pull request merged**